### ISSUE TITLE
[Bug]: `bal doc` command fails in module-ballerina-graphql

### ISSUE THREAD

**Issue by NipunaMadhushan** [ballerina-platform/ballerina-lang#42039]
### Description

`bal doc` command fails in module-ballerina-graphql with the latest updates of `master` branch in ballerina-lang.
It works without any issue in the Update 8.

```
[2024-01-24 12:03:02,710] SEVERE {b7a.log.crash} - Cannot invoke "io.ballerina.tools.diagnostics.Location.lineRange()" because "symbol.pos" is null 
java.lang.NullPointerException: Cannot invoke "io.ballerina.tools.diagnostics.Location.lineRange()" because "symbol.pos" is null
        at io.ballerina.compiler.api.impl.symbols.BallerinaSymbol.<init>(BallerinaSymbol.java:65)
        at io.ballerina.compiler.api.impl.symbols.BallerinaObjectFieldSymbol.<init>(BallerinaObjectFieldSymbol.java:62)
        at io.ballerina.compiler.api.impl.symbols.BallerinaClassFieldSymbol.<init>(BallerinaClassFieldSymbol.java:46)
        at io.ballerina.compiler.api.impl.symbols.BallerinaClassSymbol.fieldDescriptors(BallerinaClassSymbol.java:94)
        at org.ballerinalang.docgen.generator.model.Type.fromSemanticSymbol(Type.java:404)
        at org.ballerinalang.docgen.generator.model.Type.fromSemanticSymbol(Type.java:365)
        at org.ballerinalang.docgen.generator.model.Type.lambda$fromSemanticSymbol$13(Type.java:480)
        at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
        at java.base/java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1092)
        at org.ballerinalang.docgen.generator.model.Type.fromSemanticSymbol(Type.java:480)
        at org.ballerinalang.docgen.generator.model.Type.lambda$fromSemanticSymbol$11(Type.java:461)
        at java.base/java.util.Optional.ifPresent(Optional.java:178)
        at org.ballerinalang.docgen.generator.model.Type.lambda$fromSemanticSymbol$12(Type.java:442)
        at java.base/java.util.LinkedHashMap.forEach(LinkedHashMap.java:721)
        at java.base/java.util.Collections$UnmodifiableMap.forEach(Collections.java:1553)
        at org.ballerinalang.docgen.generator.model.Type.fromSemanticSymbol(Type.java:416)
        at org.ballerinalang.docgen.generator.model.Type.fromSemanticSymbol(Type.java:365)
        at org.ballerinalang.docgen.generator.model.Type.fromNode(Type.java:158)
        at org.ballerinalang.docgen.generator.model.Type.fromNode(Type.java:220)
        at org.ballerinalang.docgen.Generator.getDefaultableVariableList(Generator.java:824)
        at org.ballerinalang.docgen.Generator.getFunctionModel(Generator.java:699)
        at org.ballerinalang.docgen.Generator.getClassModel(Generator.java:523)
        at org.ballerinalang.docgen.Generator.setModuleFromSyntaxTree(Generator.java:139)
        at org.ballerinalang.docgen.docs.BallerinaDocGenerator.getDocsGenModel(BallerinaDocGenerator.java:499)
        at org.ballerinalang.docgen.docs.BallerinaDocGenerator.generateAPIDocs(BallerinaDocGenerator.java:195)
        at io.ballerina.cli.task.CreateDocsTask.execute(CreateDocsTask.java:60)
        at io.ballerina.cli.TaskExecutor.executeTasks(TaskExecutor.java:40)
        at io.ballerina.cli.cmd.DocCommand.execute(DocCommand.java:189)
        at java.base/java.util.Optional.ifPresent(Optional.java:178)
        at io.ballerina.cli.launcher.Main.main(Main.java:58)
```

### Steps to Reproduce

1. Clone `module-ballerina-graphql` module and build the module.
2. Navigate to `/ballerina` directory and run `bal doc` command.

### Related area

-> Compilation

**Comment by NipunaMadhushan** [comment]
Issue comes when we are going generate docs for https://github.com/ballerina-platform/module-ballerina-graphql/blob/master/ballerina/node_modifier_context.bal#L20.

The reason for the above is that the symbol position is `null` for the above mentioned field. The error comes when we use `fieldDescriptor()` method.

https://github.com/ballerina-platform/ballerina-lang/blob/master/compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaClassSymbol.java#L85

**Comment by HindujaB** [comment]
We set the position information to `null` to fix this issue https://github.com/wso2-enterprise/internal-support-ballerina/issues/353. As the field assignments are part of the type descriptors and conceptually not execution points, we removed the positions from them. We pass `null` as the `pos` value for many other places in the compiler when the nodes are ballerina-generated ones, therefore they are not considered as execution lines at runtime. (Eg. : https://github.com/ballerina-platform/ballerina-lang/blob/859fff4f323c44738aa7115d7aa29997f2b72cc8/compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java#L1160)

Is it possible to handle the null in the `BallerinaSymbol` itself? As there is a possibility of getting NPE from other nodes as well.
